sample_consensus/include/pcl/sample_consensus/impl/sac_model_sphere.hpp

```diff
@@ -414,18 +412,19 @@ pcl::SampleConsensusModelSphere<PointT>::doSamplesVerifyModel (
     return (false);
   }
 
+  const float sqr_inner_radius = (model_coefficients[3] <= threshold ? 0.0f : (model_coefficients[3] - threshold) * (model_coefficients[3] - threshold));
+  const float sqr_outer_radius = (model_coefficients[3] + threshold) * (model_coefficients[3] + threshold);
   for (const auto &index : indices)
   {
-    // Calculate the distance from the point to the sphere as the difference between
-    //dist(point,sphere_origin) and sphere_radius
-    if (std::abs (sqrt (
-                    ( (*input_)[index].x - model_coefficients[0] ) *
-                    ( (*input_)[index].x - model_coefficients[0] ) +
-                    ( (*input_)[index].y - model_coefficients[1] ) *
-                    ( (*input_)[index].y - model_coefficients[1] ) +
-                    ( (*input_)[index].z - model_coefficients[2] ) *
-                    ( (*input_)[index].z - model_coefficients[2] )
-                   ) - model_coefficients[3]) > threshold)
+    // To avoid sqrt computation: consider one larger sphere (radius + threshold) and one smaller sphere (radius - threshold).
+    // Valid if point is in larger sphere, but not in smaller sphere.
+    const float sqr_dist = ( (*input_)[index].x - model_coefficients[0] ) *
```

Comment:
Something like `((*input_)[index].getVector3fMap () - model_coefficients).squaredNorm ()`? Yes that would make it look nicer. Alternatively I could write `Eigen::Vector4f p ((*input_)[index].x, (*input_)[index].y, (*input_)[index].z, 0.0);` and a center like in normal_sphere.hpp, to utilize Eigen's SIMD capabilities

Reply:
You can get a Vector4f map too from a point. No need to construct another vector. The extra data member is "hidden" and is 0 (unless someone modified it).

Reply:
Since there is no guarantee that it is 0, it could mess up the computation. I think somewhere I saw it being set to 1. It is safer to not rely on that and consider it undefined. That is probably why in [normal sphere a new vector is constructed](https://github.com/PointCloudLibrary/pcl/blob/master/sample_consensus/include/pcl/sample_consensus/impl/sac_model_normal_sphere.hpp#L80).

Reply:
I went with the `getVector3fMap`-variant, because the `Vector4f` only has an advantage if SSE is available, and then the specialized SSE function is used anyway.